infrastructures/openshift/pom.xml

```diff
@@ -95,6 +95,10 @@
             <groupId>org.eclipse.che.core</groupId>
             <artifactId>che-core-commons-inject</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.che.core</groupId>
+            <artifactId>che-core-commons-observability</artifactId>
```

Comment:
I'm worried about the chicken-egg problem. I think the dependencies graph should be the opposite.

Reply:
Yeah, but that module deals with low-level constructs like thread pools, etc, hence its place in "commons", because it is generally reusable. What you're adding there is coupled to quite a number of concepts that are specific to workspaces running in kubernetes. So given our general placement of "common" and "wsmaster" specific things, IMHO, the log watching logic belongs under wsmaster, where `che-core-api-metrics` is the place responsible for workspace metric collection - which is what you're adding in this PR.

Reply:
yes of course. My point was that that comment and commit contradicts themselves, which is confusing.

Also you mentioned "commons" and "wsmaster", which is imho exactly what we should have instead of the humongous maven modules structure. We're totally overusing maven modules over java packages without any technical benefits (like independent deployment) and causing massive noise for human brain. But maybe I'm the only one. But that's different discussion. I rather close this thread :D